### PR TITLE
chore: add check for context cancel or timeout

### PR DESCRIPTION
**Pull request by pgautier404**

### CODE REVIEW

File: incubating/simple_cache_client_test.go [hunk 1]
```diff
@@ -33,121 +87,141 @@ func TestBasicHappyPathLocalPubSub(t *testing.T) {
...
+		publishTopic(client, i, ctx)
+		// Call the cancel function here and make sure it stops the messages.
+		if i == cancelAtNumber {
+			cancelFunction()
 		}
 		time.Sleep(time.Second)
 	}
+
+	// Ensure cancelFunction is called to quiet a context leak warning
+	cancelFunction()
```
Comment (eaddingtonwhite) [inline incubating/simple_cache_client_test.go:117]: nit I think this is ok to leave as is. One thing you could do is right when cancelFunction is defined could call 
`defer cancelFunction()` to ensure this is called when this overall function exits

**Review by eaddingtonwhite — APPROVED** [review]
Nice @pgautier404 Thanks for jumping on this!

File: incubating/simple_cache_client_test.go (hunk 1 above)
Comment (pgautier404) [reply incubating/simple_cache_client_test.go:117]: Ah, yes, gotcha. I'll make that change in a followup PR, thanks!

Comment (eaddingtonwhite) [reply incubating/simple_cache_client_test.go:117]: yeah sounds good!

**Pull request merged**